library/src/main/java/com/novoda/downloadmanager/VersionOneToVersionTwoMigrator.java

```diff
@@ -65,12 +81,28 @@ public void migrate() {
 
         Log.d(TAG, "all data migrations are COMMITTED, about to delete the old database, time is " + System.nanoTime());
 
-        migrationCallback.onUpdate("Deleting Database");
+        internalMigrationStatus.markAsDeleting();
+        notifyOfProgress(internalMigrationStatus);
         Log.d(TAG, "all traces of v1 are ERASED, time is " + System.nanoTime());
         database.close();
 
         database.deleteDatabase();
-        migrationCallback.onUpdate("Migration Complete");
+        internalMigrationStatus.markAsComplete();
+        notifyOfProgress(internalMigrationStatus);
+    }
+
+    private void notifyOfProgress(MigrationStatus migrationStatus) {
+        String channelName = channelCreator.getNotificationChannelName();
+        NotificationInformation notification = notificationCreator.createNotification(channelName, migrationStatus);
+
+        migrationService.updateMessage(migrationStatus);
+
+        if (migrationStatus.status() == MigrationStatus.Status.COMPLETE) {
```

Comment:
I've been torn as to whether just pass generic `Notification` back to the `activity` or to pass back `MigrationStatus` and let the `Service` decide which type of notification to use based on that. What do you think?

Reply:
I might be torn to use the most obvious solution for the client (Activity)

Reply:
I don't understand your comment @zegnus 😄

Reply:
I think I put `Activity` in the comment by accident earlier 😄 the `Service` is the one that creates and displays the notification. We have an optional callback for the activity to receive `NotificationStatus` and act on that.